include/deal.II/lac/petsc_snes.templates.h

```diff
@@ -543,7 +545,6 @@ namespace PETScWrappers
 
 
         // Do not waste memory by creating a dummy AIJ matrix inside PETSc.
```

Comment:
Should this comment (`// Do not waste memory by creating a dummy AIJ matrix inside PETSc.`) go with the `this->need_dummy_assemble = false` statement? If yes, what about we remove the comment?